src/Event.cc

```diff
@@ -97,27 +97,63 @@ Event::Event() {
 void Event::_init() {
     _keywords.insert(TYPE);
     _keywords.insert(EVENTTIME);
-    _keywords.insert(HOSTID);
-    _keywords.insert(RUNID);
+//    _keywords.insert(RUNID);
     _keywords.insert(STATUS);
     _keywords.insert(TOPIC);
     _keywords.insert(PUBTIME);
 }
 
 Event::Event(cms::TextMessage *msg) {
-    _init();
+    // _init();
+
+    vector<std::string>names = msg->getPropertyNames();
+    unsigned int i;
```

Comment:
You're not using this except in the `for` loops below, so just define it in the loops.